src/components/common/CommuteModal.tsx

```diff
@@ -0,0 +1,141 @@
+import { useEffect, useRef, useState } from "react";
+import Button from "./Button";
+import * as style from "./commuteModalStyle";
+import { doc, setDoc } from "firebase/firestore";
+import { db } from "@/firebase/firebase";
+
+interface Props {
+  showModal: boolean;
+  setShowModal: React.Dispatch<React.SetStateAction<boolean>>;
+  startTime: Date | null;
+  setStartTime: React.Dispatch<React.SetStateAction<Date | null>>;
+  endTime: Date | null;
+  setEndTime: React.Dispatch<React.SetStateAction<Date | null>>;
+  workingHours: number;
+  workingMinutes: number;
+  setWorkedHours: React.Dispatch<React.SetStateAction<number>>;
+  setWorkedMinutes: React.Dispatch<React.SetStateAction<number>>;
+  disabledEndBtn: boolean;
+  setDisabledEndBtn: React.Dispatch<React.SetStateAction<boolean>>;
+  uid: string | undefined;
```

Comment:
```suggestion
  uid?: string;
```
으로 적지 않으신 이유가 있나요? 기존 방식은 undefined 라도 꼭 값을 넘겨줘야 합니다.

Reply:
uid가 지금 전역으로 사용되고 있어서 계속 props로 전달 받고 있죠..? 이런걸 props drilling 이라고 합니다. 
해결할 수 있는 많은 방법들이 있는데, 

- context api
- 전역 상태 관리 툴
- 브라우저 저장소

중 마음에 드시는 걸로 활용해보세요, props 가 끊임없이 전달되는 현상을 막고 가독성과 DX을 높혀줄겁니다.

Reply:
시간이 부족해서 뭐가 더 적절한 코드인지 고민해보지 않고 그냥 에러가 사라지면 넘어갔던 것 같습니다...
다음 리펙토링, 다음 프로젝트에서는 고려해서 작업하겠습니다... 감사합니다!